do not autocomplete when i am in the whitespace before a line

markijbema opened the pull request:
## Context

when indenting autocomplete would autocomplete, and then indenting further with tab would accept the change. This PR fixes that.

Before:
![image](https://github.com/user-attachments/assets/bd24cee5-feda-401d-8072-58eaabc49787)

After:
![image](https://github.com/user-attachments/assets/bbcf6c16-b231-476d-a063-b2776daa17ef)

kamilchm committed:
Support project id set in env.GOOGLE_CLOUD_PROJECT for Gemini CLI

This is needed for using Code Assist subscription.

Commit:
do not autocomplete when i am in the whitespace before a line

markijbema commented on src/package.json:
```diff
@@ -262,6 +262,11 @@
+			{
+				"command": "editor.action.inlineSuggest.commit",
+				"key": "tab",
+				"when": "inlineSuggestionVisible && editorTextFocus && !editorTabMovesFocus && !inSnippetMode && !suggestWidgetVisible && editorLangId != 'markdown'"
```
I'm not sure whether this is the best solution, this is what sonnet thought up for enabling me to accept suggestions on empty lines with tab. I was kind of surprised tab didn't work on lines without anything on it tbh.

chrarnoldus replied:
What does this do? What if multiple extensions try to do this?

Commit:
add changeset

Commit:
mark files as kilo only

chrarnoldus committed:
Remove kilocode_change marker

https://github.com/RooCodeInc/Roo-Code/pull/5342

kevinvandijk committed:
Add changeset

hassoncs reviewed (approved):
This mostly looks fine to me! My big refactor PR here adds some test utils that make some of this mockDocument stuff a little easier in the future! https://github.com/Kilo-Org/kilocode/pull/602/files#diff-66e1d8e757dc49cf8fa0d71fd8fc02f14482f526ee7f43a3432e3bb3a3928639

I'm going to approve this because I think it makes things strictly better than what we have today! We can rebase and split out my PR to get things ready to review after

hassoncs committed:
ci: Hardcode the chromatic token in chromatic.yml (#968)

Our chromatic token is already public, so let's just hardcode it in our chromatic workflow so the chromatic steup doesn't fail for non-kilo users.

Co-authored-by: Chris Hasson <noreply@example.com>

github-actions[bot] committed:
changeset version bump

Commit:
Updating CHANGELOG.md format

markijbema committed:
Update copilot-instructions.md

chrarnoldus committed:
Preserve focus when opening diff view

Works for me!

chrarnoldus committed:
Fix tests

chrarnoldus committed:
Show error message when not logged in with Kilo Code provider

github-actions[bot] committed:
changeset version bump

Commit:
Updating CHANGELOG.md format

hassoncs committed:
refactor(git): exclude lock files from commit generation diff (#942)

* Introduces an exclusionUtils.ts file to define a comprehensive list of common lock files across various programming languages and ecosystems.

The getStagedDiff method in GitExtensionService is updated to automatically exclude these lock files when generating the staged diff. This prevents large, often irrelevant, changes in lock files from cluttering the diff output used for commit message generation, improving the relevance and accuracy of the generated messages.

A new test file GitExtensionService.spec.ts is added to verify that getStagedDiff correctly applies the exclusion logic for lock files.

This change addresses the issue of overly large diffs caused by frequently changing lock files, which can hinder effective commit message generation.

Fixes: #924

* refactor(git): Use the ignore library to figure out files to ignore for commit message generation

Previously we were doing string matching ourselves, this updates the ignore code to use the `ignored` library which is what the RooIgnoreControll uses.

* Check for ignored files using the RooIgnoreController before generating diffs for commit generation

* refactor(git): Remove redundant error handling in GitExtensionService

Remove comments and try/catches we don't need

* refactor(git): Ensure workspaceRoot is always defined

* refactor(git): Migrate git command execution to `spawnSync`

By using `spawnSync`, we get the benefit of Node's automatic file name escaping for paths, so we don't have to maintain any special character escaping logic for different platforms.

---------

Co-authored-by: Chris Hasson <noreply@example.com>

catrielmuller committed:
refactor: show full user prompt on logs
